### PR TITLE
corp: use thread pool to execute pushdown

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @siddontang @disksing PTAL

### CODE REVIEW

File: src/server/coprocessor.rs [hunk 1]
```diff
@@ -86,105 +88,33 @@ impl From<txn::Error> for Error {
...
-        if let Err(e) = self.handle.take().unwrap().join() {
-            error!("failed to stop the coprocessor: {:?}", e);
-        }
+    pub fn on_request(&self, req: Request, token: Token, msg_id: u64) {
+        let end_point = self.snap_endpoint.clone();
+        let ch = self.ch.clone();
+        self.pool.execute(move || {
+            let timer = Instant::now();
+            end_point.handle_request(req, token, msg_id, ch);
+            debug!("request {:?}/{} takes {:?}", token, msg_id, timer.elapsed());
```
Comment (siddontang) [inline src/server/coprocessor.rs]: this may cause too many logs, we can only record slow handle or use trace.

**Comment by siddontang** [comment]
Rest LGTM

**Comment by ngaut** [comment]
LGTM

**Comment by BusyJay** [comment]
@siddontang PTAL

**Comment by siddontang** [comment]
LGTM

**Pull request merged**